WildMeOrg/codex-frontend
# Dex 1034 fix sex select and disambiguate duplicate custom field labels

This PR solves a few issues:

1. Fixes some bugs with multi-select editor components that would crash when you tried to add more options to customFields that are multi-select in Control Panel → Manage Fields.

2. Removes src/models/assetGroup/useAssetGroupSchemas.js, which doesn't seem to get used anywhere.

3. Fixes a bug in the SaveField component where it should have been getting defaultValue from the schema instead of default from the formData object directly.

4. Fixes a bug where sex was a string field rather than a select field. Now, as a select field, bulk import entries of sex can be manually mapped to correct sex values during the bulk import process.

5. Fixes a bug where custom fields that were designated as select or multi-select were not being treated as such.

This PR does not solve the issue where multi-select custom fields can't be manually mapped to multiple values (see https://wildme.atlassian.net/browse/DEX-1080 for clarification about this issue. It has been separated out for the time being).
<img width="1552" alt="Screen Shot 2022-05-23 at 5 02 02 PM" src="https://user-images.githubusercontent.com/2775448/169922905-8dbd984d-0b30-4780-acae-15f794e27b51.png">
<img width="1552" alt="Screen Shot 2022-05-23 at 5 03 19 PM" src="https://user-images.githubusercontent.com/2775448/169922997-2eed5ba9-43df-407c-bd5d-d8941404f556.png">


QA Notes:
After merge, you should now be able to: 
1. Designate sex manually during bulk import correctly (try maybe mis-spelling sex in  your bulk imports to test this).
2. Add options to multi-select and select customFields (e.g., Behavior at the encounter and at the sighting level).
3. See a distinction between custom fields that have the same label. For instance, zebra has a label, "Behavior" that is used for both the custom field at the encounter level and separately for a custom field at the sighting level. Previously, it would have been impossible to know which of the two "Behavior"s you were assigning to a given column during bulk import:
<img width="444" alt="Screen Shot 2022-05-23 at 5 07 39 PM" src="https://user-images.githubusercontent.com/2775448/169923326-5f5cf3de-9564-4804-974d-59883f917d45.png">
